src/components/server/ServerManager.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  CommandBar,
+  ICommandBarItemProps,
+  INavLink,
+  Nav,
+  PrimaryButton,
+  Stack,
+} from "@fluentui/react";
+import React, { FC, useEffect, useState } from "react";
+import Server, { SERVER_PREFIX } from "../../types/server";
+import { getItems, saveItem, deleteItem } from "../../storage/storage";
+import ServerEdit from "./ServerEdit";
+import ServerDeleteConfirmation from "./ServerDeleteConfirmation";
+import ServerConnectionError from "./ServerConnectionError";
+import { stackTokens, serverPanel } from "../../style";
+
+type ServerManagerProps = {
+  onConnect: (server: Server) => Promise<void>;
+};
+
+function replaceAtIndex<T>(array: T[], item: T, index: number) {
```

Comment:
Not a big one but I'd be inclined to split these out into some kind of utils module so we can use them elsewhere if needed as well

Reply:
Good idea - done 👍